src/net_processing.cpp

```diff
@@ -2837,6 +2837,18 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             nCMPCTBLOCKVersion = 1;
             m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, fAnnounceUsingCMPCTBLOCK, nCMPCTBLOCKVersion));
         }
+
+        LOCK(cs_main);
+        const auto peer_id = pfrom.GetId();
+        if (!State(peer_id)->m_wtxid_relay) {
+            // We could have optimistically pre-registered the peer on sending SENDRECON, or
+            // registered upon receiving SENDRECON afterwards.
+            // Now we are sure they won't announce WTXIDRELAY (can't happen after VERACK).
+            // We should clear the reconciliation state of the peer (because we support
+            // reconciliations only based on wtxids).
+            m_reconciliation.ForgetPeer(peer_id);
+        }
```

Comment:
in 77d1aaab3e62aae2319860049781f4e9f0087521:

I like that there is no longer an ordering requirement between wtxidrelay and sendrecon :+1: 

Question: shouldn't there be a gate for whether or not we're doing reconciliations at all?